core/src/cluster_slots.rs

```diff
@@ -42,7 +40,20 @@ impl ClusterSlots {
                 self.insert_node_id(*slot, epoch_slots.from);
             }
         }
-        self.cluster_slots.write().unwrap().retain(|x, _| *x > root);
+        let old_slots: Vec<_> = self
+            .cluster_slots
+            .read()
+            .unwrap()
+            .keys()
+            .take_while(|slot| **slot <= root)
+            .copied()
+            .collect();
+        {
+            let mut cluster_slots = self.cluster_slots.write().unwrap();
+            for slot in old_slots {
+                cluster_slots.remove(&slot);
+            }
+        }
```

Comment:
is a `split_off` followed by a `swap` more efficient here? i.e. something like: https://github.com/solana-labs/solana/blob/master/core/src/optimistic_confirmation_verifier.rs#L30-L34

Reply:
Doesn't seem different, but I switched to `split_off` anyways. thanks